utils/backoff.go

```diff
@@ -7,47 +7,32 @@ import (
 	"time"
 )
 
-const (
-	backoffBase uint64 = 128
-	maxAttempts        = 4
-)
-
-// BackoffOperation is executed by Retry.
-// The BackoffOperation will be retried the provided
-// number attempts if returning an error.
-type BackoffOperation func() error
+var backoffTimeouts = []time.Duration{50 * time.Millisecond, 100 * time.Millisecond, 200 * time.Millisecond, 200 * time.Millisecond, 400 * time.Millisecond, 400 * time.Millisecond}
 
 // ProgressiveRetry executes a BackoffOperation and retries the operation 3 times upon error.
-func ProgressiveRetry(bo BackoffOperation) error {
+func ProgressiveRetry(operation func() error) error {
 	var t *time.Timer
-	var attempt uint64
+	var attempts = 0
 
 	for {
-		err := bo()
+		err := operation()
 		if err == nil {
 			return nil
 		}
 
-		nextRetry := NextRetry(attempt)
+		attempts++
+		if attempts >= len(backoffTimeouts) {
+			return err
+		}
+
+		nextRetry := backoffTimeouts[attempts]
```

Comment:
The first time we index into `backoffTimeouts`, `attempts == 1`, so we never actually wait 50ms: always 100ms. See above for a simpler suggestion that just iterates `attempts` starting from `0`.